tests/sql_mode/data/db1.prepare.sql

```diff
@@ -0,0 +1,42 @@
+set @@session.SQL_MODE='';
+
+drop database if exists `sql_mode`;
+create database `sql_mode`;
+use `sql_mode`;
+CREATE TABLE `t_1` (
+  `id` int NOT NULL AUTO_INCREMENT,
+  `name` varchar(60),
+  `num` int,
+  `dt` datetime,
+  PRIMARY KEY (id)
+);
+
+-- test sql_mode PIPES_AS_CONCAT
+insert into t_1(num) values('pipes'||'as'||'concat');
+
+-- test sql_mode ANSI_QUOTES
+insert into t_1(name) values("a");
+
+-- test sql_mode IGNORE_SPACE
+create table count (id int not null, primary key(id));
+
+-- test sql_mode NO_AUTO_VALUE_ON_ZERO
+insert into t_1(id, name) values (10, 'a');
+insert into t_1(id, name) values (0, 'b');
+insert into t_1(id, name) values (0, 'c');
```

Comment:
since we didn't set sql mode in this SQL file, above two insert are converted to auto increment value, so the binlog is writtern with the auto increment value not zero, and this didn't test DM will add NO_AUTO_VALUE_ON_ZERO with `AdjustSQLModeCompatible`

since db2 (source2) has done this test, maybe we could remove above lines.

Reply:
Yes, I think it just tested what you said above.